src/solver.ts

```diff
@@ -36,32 +36,84 @@ const parseEnvYml = (envYml: string) => {
 const solve = async (
   specs: Array<string>,
   channels: Array<string>,
+  installedCondaPackages: ISolvedPackages,
   logger?: ILogger
 ) => {
   let result: any = undefined;
   const solvedPackages: ISolvedPackages = {};
   try {
+    let installed: any = [];
+    if (Object.keys(installedCondaPackages).length) {
+      Object.keys(installedCondaPackages).map((filename: string) => {
+        const installedPkg = installedCondaPackages[filename];
+        let tmpPkg = {
+          url: installedPkg.url,
+          packageName: installedPkg.name,
+          build: installedPkg.build_string,
+          repoName: installedPkg.repo_name,
+          filename: filename,
+          version: installedPkg.version,
+          depends: installedPkg?.depends,
+          buildNumber: installedPkg.build_number,
+          subdir: installedPkg?.subdir
+        };
+
+        installed.push(tmpPkg);
+      });
+    } else {
+      installed = undefined;
+    }
     const startSolveTime = performance.now();
-    result = await simpleSolve(specs, channels, PLATFORMS);
+    result = await simpleSolve(specs, channels, PLATFORMS, installed);
     const endSolveTime = performance.now();
-
     if (logger) {
       logger.log(
         `Solving took ${(endSolveTime - startSolveTime) / 1000} seconds`
       );
     }
 
-    result.map((item: any) => {
-      const { filename, packageName, repoName, url, version, build } = item;
-      solvedPackages[filename] = {
-        name: packageName,
-        repo_url: repoName,
-        build_string: build,
-        url: url,
-        version: version,
-        repo_name: repoName
-      };
-    });
+    if (!result.length) {
+      logger?.log(`${specs.join(',')} have been already installed`);
+    } else {
+      specs.map((spec: string) => {
+        let isIncluded = false;
+        result.map((pkg: any) => {
+          if (spec.includes(pkg.packageName)) {
+            isIncluded = true;
+          }
+        });
+        if (!isIncluded) {
+          logger?.log(
+            `Conditions for ${spec} were met before and the package was already installed`
+          );
+        }
+      });
```

Comment:
Comparing specs with package name in `spec.includes(pkg.packageName)` will not work properly. e.g. if spec is `jupyterlab-unfold` and package name is `jupyterlab`, the spec includes the package name, but it doesn't make sense to consider them the same package.

I'd suggest simply removing these logs for now.

```suggestion
```